### PR TITLE
Commerce 8133

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
As a Sales Manager

I want to limit the ability of my customers to create new or edit existing shipping address set against an order

So that I can limit the potential for fraud or theft

https://issues.liferay.com/browse/COMMERCE-8133

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp [hunk 1]
```diff
@@ -40,34 +40,40 @@ if (commerceOrder.isGuestOrder()) {
...
+CommerceAccount commerceAccount = commerceContext.getCommerceAccount();
 %>
 
 <div class="form-group-autofit">
 	<c:if test="<%= !commerceOrder.isGuestOrder() %>">
-		<aui:select label="<%= selectLabel %>" name="commerceAddress" onChange='<%= liferayPortletResponse.getNamespace() + "selectAddress();" %>' wrapperCssClass="commerce-form-group-item-row form-group-item">
-			<aui:option label="add-new-address" value="0" />
+		<c:if test="<%= baseAddressCheckoutStepDisplayContext.hasPermission(permissionChecker, commerceAccount, CommerceAccountActionKeys.VIEW_ADDRESSES) %>">
+			<aui:select label="<%= selectLabel %>" name="commerceAddress" onChange='<%= liferayPortletResponse.getNamespace() + "selectAddress();" %>' wrapperCssClass="commerce-form-group-item-row form-group-item">
+				<c:if test="<%= baseAddressCheckoutStepDisplayContext.hasPermission(permissionChecker, commerceAccount, CommerceAccountActionKeys.MANAGE_ADDRESSES) %>">
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp:51]: We should replace CommerceAccountActionKeys.MANAGE_ADDRESSES with the one added to portal's Account module

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/BaseAddressCheckoutStepDisplayContext.java [hunk 2]
```diff
@@ -70,6 +80,22 @@ public CommerceOrder getCommerceOrder() {
 
 	public abstract String getTitle();
 
+	public boolean hasPermission(
+			PermissionChecker permissionChecker,
+			CommerceAccount commerceAccount, String actionId)
+		throws PortalException {
+
+		if (commerceAccount.isPersonalAccount() ||
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/BaseAddressCheckoutStepDisplayContext.java:88]: Add check for guest type account, should be managed like the peronsal one

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]
check comments

File: modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/BaseAddressCheckoutStepDisplayContext.java (hunk 2 above)
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/BaseAddressCheckoutStepDisplayContext.java:88]: This is making some tests to fail

**Comment by luca-pellizzon** [comment]
see https://github.com/liferay-commerce/liferay-portal/pull/2163

**Pull request closed without merging**